client/src/partials/primary_cash/expense/generic.js

```diff
@@ -33,196 +90,126 @@ angular.module('bhima.controllers')
-          },
-          'creditor_group' : {
-            columns : ['account_id']
-          }
-        },
-        join : ['supplier.creditor_uuid=creditor.uuid', 'creditor.group_uuid=creditor_group.uuid']
-      }
-    };
-
-    dependencies.currencies = {
-      query : {
-        tables : {
-          'currency' : {
-            columns : ['id', 'name', 'symbol']
-          }
-        }
-      }
-    };
-
-    dependencies.accounts = {
-      query : {
-        tables : {
-          'account' :{
-            columns : ['id', 'account_txt', 'account_number', 'is_ohada']
-          }
-        },
-        where : ['account.is_ohada=1', 'AND', 'account.account_type_id<>3']
-      }
-    };
+  function getAccount (ac) {
+    if (!ac) { return; }
+     session.configured = true;
+     session.ac = ac;
+     session.complete = true;
+  }
 
-    cache.fetch('currency').then(load);
-    cache.fetch('account').then(getAccount);
+  function generate () {
+    session.receipt.reference_uuid = uuid();
+  }
 
-    function load (currency) {
-      if (!currency) { return; }
-      session.currency = currency;
-      haltOnNoExchange();
-    }
+  function clear () {
+    session.receipt = {};
+    session.receipt.date = new Date();
+    session.receipt.value = 0.00;
+    session.receipt.cash_box_id = $routeParams.id;
+  }
 
-    function getAccount (ac) {
-      if (!ac) { return; }
-       session.configured = true;
-       session.ac = ac;
-       session.complete = true;
+  function valid () {
+    if (!session || !session.receipt) {
+      session.invalid = true;
+      return;
     }
+    var r = session.receipt;
+
+    session.invalid = !(isDefined(session.currency) &&
+      isDefined(r.cost) &&
+      r.cost > 0 &&
+      isDefined(r.description) &&
+      isDefined(r.date) &&
+      isDefined(r.cash_box_id));
+  }
 
-    appstate.register('project', function (project) {
-      $scope.project =  project;
-      validate.process(dependencies)
-      .then(function (models) {
-        angular.extend($scope, models);
-        session.receipt.date = new Date();
-        session.receipt.cost = 0.00;
-        session.receipt.cash_box_id = $routeParams.id;
-
-        session.accounts = models.accounts.data;
-      })
-      .catch(function (err) {
-        messenger.danger(err);
-      });
-    });
-
-    $scope.generate = function generate () {
-      session.receipt.reference_uuid = uuid();
-    };
-
-    $scope.clear = function clear () {
-      session.receipt = {};
-      session.receipt.date = new Date();
-      session.receipt.value = 0.00;
-      session.receipt.cash_box_id = $routeParams.id;
-    };
+  function submit () {
+    var data, receipt = session.receipt;
+
+    connect.fetch('/user_session')
+    .then(function (user) {
+
+      data = {
+        uuid          : uuid(),
+        reference     : 1,
+        project_id    : $scope.project.id,
+        type          : 'E',
+        date          : util.sqlDate(receipt.date),
```

Comment:
You can just send back a date object, no need to use `util.sqlDate()`.  It isn't responsible for our date issues.